File: app/src/main/java/com/booisajerk/typewritersamples/TypewriterDetailActivity.java
package com.booisajerk.typewritersamples;

import android.content.Intent;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

public class TypewriterDetailActivity extends AppCompatActivity {

    //TODO display app bar

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_typewriter);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        Intent intent = new Intent(TypewriterDetailActivity.this, MainActivity.class);
        intent.getExtras();

        //TODO get the intent from MainActivity
        Typeface typewriterFont = Typeface.createFromAsset(getAssets(), "fonts/lc_smith_5_typewriter.ttf");

        TextView detailTitle = (TextView) findViewById(R.id.detail_title);
        detailTitle.setTypeface(typewriterFont);

        TextView detailText = (TextView) findViewById(R.id.detail_text);
        detailText.setTypeface(typewriterFont);

        ImageView detailImage = (ImageView) findViewById(R.id.detail_image);

        //TODO add case statement
    }
}

File: app/src/main/java/com/booisajerk/typewritersamples/TypewriterDetailActivity.kt
package com.booisajerk.typewritersamples

import android.content.Context
import android.content.Intent
import android.graphics.Typeface
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import kotlinx.android.synthetic.main.content_typewriter.*

class TypewriterDetailActivity : BaseActivity() {

    lateinit var machine: Machine

    companion object {
        val TAG = "Parker" + TypewriterDetailActivity::class.qualifiedName
        val NAVIGATION_ID = "navigationId"

        fun detailIntent(context: Context, position: Int): Intent {
            val intent = Intent(context, TypewriterDetailActivity::class.java)
            intent.putExtra(NAVIGATION_ID, position)
            return intent
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_typewriter)
        Log.d(TAG, "onCreate")

        loadMachine()
    }

    private fun loadMachine() {
        machine = MachineData.machineList()[intent.getIntExtra(NAVIGATION_ID, 0)]
        detail_image.setImageResource(machine.getImageResourceId(this))
        detail_title.setText(machine.name)
        setTypewriterFont(detail_title)
        setTypewriterFont(detail_text)
    }

    private fun setTypewriterFont(text:TextView) {
        val typewriterFont = Typeface.createFromAsset(assets, "fonts/lc_smith_5_typewriter.ttf")
        text.typeface = typewriterFont
    }
}

